src/napari/_vispy/canvas.py

```diff
@@ -875,41 +904,113 @@ def _update_layer_overlays(self, layer: Layer) -> None:
 
             self._add_layer_overlay(layer, overlay, parent)
 
-        self._update_overlay_canvas_positions()
+        self._defer_overlay_position_update()
 
-    def _get_ordered_visible_canvas_overlays(self):
+    def _get_ordered_visible_canvas_overlays(
+        self,
+    ) -> Iterator[tuple[CanvasOverlay, VispyBaseOverlay, Node | None]]:
         # note that some canvas overlays do no use CanvasPosition, but are instead
         # free-floating (such as the cursor overlay), so those are skipped
 
-        # first viewer overlays
+        # first the base view (non-gridded viewer overlays)
         for overlay, vispy_overlays in self._overlay_to_visual.items():
             if (
                 overlay.visible
                 and isinstance(overlay, CanvasOverlay)
                 and overlay.position in list(CanvasPosition)
```

Comment:
3 times we do this filtering in this function. 
Maybe we should extract it to variable and just use?